cmd/dex-worker/main.go

```diff
@@ -48,6 +48,7 @@ func main() {
 
 	enableRegistration := fs.Bool("enable-registration", false, "Allows users to self-register")
 	enableClientRegistration := fs.Bool("enable-client-registration", false, "Allow dynamic registration of clients")
+	registerOnFirstLogin := fs.Bool("register-on-first-login", false, "When a user logs in through a federated identity service, automatically register them if they don't have an account. This flag is independent of enable-registration.")
```

Comment:
> enableRegistration only affects if the register screen will be shown

That's not all it does: much more importantly, it turns off the /registration endpoint. If it was just a matter of the register page being shown or not, that would be a huge security hole.